src/libs/models/BankAccount.ts

```diff
@@ -0,0 +1,261 @@
+import Str from 'expensify-common/lib/str';
+import CONST from '../../CONST';
+import { ValueOf } from "type-fest";
+
+type State = ValueOf<typeof BankAccount.STATE>;
+
+type BankAccountType = "withdrawal" | "direct-deposit";
+
+type BankAccountJSON = {
+    methodID: number;
+    validateCodeExpectedDate: string;
+    isSavings: boolean;
+    accountData: {
+        accountNumber: string;
+        addressName: string;
+        processor: string;
+        routingNumber: string;
+        sharees: string[];
+        state: State;
+        country: string;
+        defaultCredit: boolean;
+        allowDebit: boolean;
+        riskChecked: boolean;
+        additionalData: Record<string, unknown>;
+    }
+};
+
+type ACHData = {
+    routingNumber: string,
+    accountNumber: string,
+    addressName: string,
+    isSavings: boolean,
+    bankAccountID: number,
+    state: State,
+    validateCodeExpectedDate: string,
+    needsToUpgrade: boolean,
+};
+
+class BankAccount {
+    public static readonly STATE = {
+        PENDING: 'PENDING',
+        OPEN: 'OPEN',
+        DELETED: 'DELETED',
+        LOCKED: 'LOCKED',
+        SETUP: 'SETUP',
+        VERIFYING: 'VERIFYING',
+    };
+    private readonly json: BankAccountJSON;
+    constructor(accountJSON: BankAccountJSON) {
+        this.json = accountJSON;
+    }
+
+    /**
+     * @returns - the ID of the reimbursement account
+     */
+    getID() {
```

Comment:
```suggestion
    getID(): number {
```

Could you add return types for all functions here (except if isn't returning anything)?